source/crud/tests/unified/insertOne-dots_and_dollars.yml

```diff
@@ -0,0 +1,241 @@
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dottedKeyInEmbedded
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKeyInEmbedded
+
+  - description: "Inserting document with dollar-prefixed key in _id yields server-side error"
+    # Note: 5.0+ did not remove restrictions on dollar-prefixed keys in _id documents
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: &dollarPrefixedKeyInId { _id: { $a: 1 } }
+        expectError:
+          isClientError: false
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dollarPrefixedKeyInId
+    outcome: *initialData
+
+  - description: "Inserting document with dotted key in _id on 3.6+ server"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: &dottedKeyInId { _id: { a.b: 1 } }
+        expectResult:
+          insertedCount: 1
+          insertedId: { $$unsetOrMatches: { a.b: 1 } }
+    expectEvents: &expectEventsDottedKeyInId
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                insert: *collection0Name
+                documents:
+                  - *dottedKeyInId
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKeyInId
+
+  - description: "Inserting document with dotted key in _id on pre-3.6 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "3.4.99"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          document: *dottedKeyInId
+        expectError:
+          isClientError: false
+    expectEvents: *expectEventsDottedKeyInId
+    outcome: *initialData
+
+  - description: "Inserting document with DBRef-like keys"
+    operations:
+      - name: insertOne
+        object: *collection0
+        arguments:
+          # Note: an incomplete DBRef document may cause issues loading the test
+          # file with an Extended JSON parser, since the presence of one DBRef
+          # key may cause the parser to require others and/or enforce expected
+          # types (e.g. $ref and $db must be strings).
+          #
+          # Using "$db" here works for libmongoc so long as it's a string type;
+          # however, neither $ref nor $id would be accepted on their own.
+          #
+          # See https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#parsers
```

Comment:
To clarify, this limitation has nothing to do with the driver's insert functionality. It's a problem with using an Extended JSON parser to load the test file itself.

I realize this comment is pretty specific to libmongoc, but I thought it was helpful to explain why I couldn't use `$ref` or `$id`. I'm hoping this doesn't cause issues for other drivers, but if so we can consider removing this test entirely if so.